config.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import yaml
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+spack_version='v0.14.1'
+
+parser=argparse.ArgumentParser(description='Small config script which can be used to install a spack instance with the correct configuration files and mch spack packages.')
+parser.add_argument('-i', '--idir', type=str, default=dir_path, help='Where the Spack instance is installed or you want it to be installed')
+parser.add_argument('-m', '--machine', type=str, help='Required: machine name')
+parser.add_argument('-v', '--version', type=str, default='v0.14.1', help='Spack version, Default: ' + spack_version)
+parser.add_argument('-r', '--reposdir', type=str, help='repos.yaml install directory')
+parser.add_argument('-p', '--pckgidir', type=str, help='Define spack package, modules & stages installation directory. Default: tsa; /scratch/$USER/spack, daint; /scratch/snx3000/$USER/spack')
+args=parser.parse_args()
+
+if args.machine is None:
+    print('Error: machine name required!')
+    exit()
+  
+if args.idir is not None:
+    if not os.path.isdir(args.idir + '/spack'):
+        print('Cloning spack instance to:' + args.idir)
+        if args.version is None:
+            args.version = spack_version
+        clone_cmd = 'git clone git@github.com:spack/spack.git -b' + args.version + ' ' + args.idir + '/spack'   
+        os.system(clone_cmd)
+
+print('Installing mch packages & ' + args.machine + ' config files')
+
+if args.reposdir is not None:
+    if os.path.isdir(args.reposdir) and not os.path.isfile(args.reposdir + '/repos.yaml'):
+        repos_data = yaml.load(open('./sysconfigs/repos.yaml', 'r+'))
+        repos_data['repos'] = args.reposdir
+        yaml.dump(repos_data, open('./sysconfigs/repos.yaml', 'r+'))
```

Comment:
uhmmm does this work? dont you need to open the file in write mode with 'w' instead of 'r+'?

Reply:
I need write and read, so r+? At least from what python is saying while compiling!

Reply:
why do you need to read? I think in the first line yaml.load you are only reading 'r' and in the second line yaml.dump you are only writing 'w', since you are dumping the entire yaml structure (not just the modified 'repos' ) again .